EventFilter/CSCRawToDigi/src/CSCDigiToRawAccept.cc

```diff
@@ -0,0 +1,83 @@
+#include "DataFormats/CSCDigi/interface/CSCConstants.h"
+#include "EventFilter/CSCRawToDigi/interface/CSCDigiToRawAccept.h"
+
+CSCDetId CSCDigiToRawAccept::chamberID(const CSCDetId& cscDetId) {
+  CSCDetId chamberId = cscDetId.chamberId();
+  if (chamberId.ring() == 4) {
+    chamberId = CSCDetId(chamberId.endcap(), chamberId.station(), 1, chamberId.chamber(), 0);
+  }
+  return chamberId;
+}
+
+// older function for pretriggers BX objects
+bool CSCDigiToRawAccept::accept(const CSCDetId& cscId,
+                                const CSCCLCTPreTriggerCollection& lcts,
+                                int bxMin,
+                                int bxMax,
+                                int nominalBX,
+                                bool me1abCheck) {
+  if (bxMin == -999)
+    return true;
+  CSCDetId chamberId = chamberID(cscId);
+  CSCCLCTPreTriggerCollection::Range lctRange = lcts.get(chamberId);
+  bool result = false;
+  for (CSCCLCTPreTriggerCollection::const_iterator lctItr = lctRange.first; lctItr != lctRange.second; ++lctItr) {
+    int bx = *lctItr - nominalBX;
+    if (bx >= bxMin && bx <= bxMax) {
+      result = true;
+      break;
+    }
+  }
+  bool me1a = cscId.station() == 1 && cscId.ring() == 4;
+  if (me1a && result == false && me1abCheck) {
+    //check pretriggers in me1a as well; relevant for TMB emulator writing to separate detIds
+    lctRange = lcts.get(cscId);
+    for (CSCCLCTPreTriggerCollection::const_iterator lctItr = lctRange.first; lctItr != lctRange.second; ++lctItr) {
+      int bx = *lctItr - nominalBX;
+      if (bx >= bxMin && bx <= bxMax) {
+        result = true;
+        break;
+      }
+    }
+  }
+  return result;
+}
+
+bool CSCDigiToRawAccept::accept(const CSCDetId& cscId,
+                                const CSCCLCTPreTriggerDigiCollection& lcts,
+                                int bxMin,
+                                int bxMax,
+                                int nominalBX,
+                                bool me1abCheck,
+                                std::vector<bool>& preTriggerInCFEB) {
```

Comment:
there seems to be some code duplication here with respect to the other `accept` methods. Is there a possibility to define some common helper functions to reduce the duplication?

Reply:
There is no duplication. One function is for ALCT, CLCT and LCT. One function is for CLCTPreTriggerBX. One function is for CLCTPreTriggerDigi.